cobigen-cli/src/main/java/com/devonfw/cobigen/cli/utils/ParsingUtils.java

```diff
@@ -48,19 +57,53 @@
      */
     public static JavaContext getJavaContext(File inputFile, File inputProject) {
 
-        JavaContext context = JavaSourceProviderUsingMaven.createFromLocalMavenProject(inputProject);
-
-        String qualifiedName = ParsingUtils.getQualifiedName(inputFile, context);
-
-        try {
-            context.getClassLoader().loadClass(qualifiedName);
-        } catch (NoClassDefFoundError | ClassNotFoundException e) {
+        JavaContext context = JavaSourceProviderUsingMaven.createFromLocalMavenProject(inputProject,true);
+        String fqn = ParsingUtils.getFQN(inputFile);
+		try {
+			context.getClassLoader().loadClass(fqn);
+		} catch (NoClassDefFoundError | ClassNotFoundException e) {
             logger.error("Compiled class " + e.getMessage()
                 + " has not been found. Most probably you need to build project " + inputProject.toString() + " .");
             System.exit(1);
         }
         return context;
     }
+    /**
+     * This method is traversing parent folders until it reaches java folder in order to get the FQN
+     * @param inputFile
+     * @return qualified name with full package
+     */
+    private static String getFQN(File inputFile) {
+        String simpleName = inputFile.getName().replaceAll("\\.(?i)java", "");
+        String packageName = getPackageName(inputFile.getParentFile(), "");
+
+         return packageName + "." + simpleName;
+    }
+/**
+ *  This method traverse the folder in reverse order from child to parent
+ *  @param folder parent input file
+ *  @param packageName 
+ *  @return package name
+ * */
```

Comment:
Please format this correctly, the JavaDoc needs to be aligned with the method declaration.